### ISSUE TITLE
changed ownership of file

### ISSUE THREAD

**Issue by rkarpuzov** [mikefarah/yq#1473]
**Describe the bug**
Ownership is changed from ejabberd to root. The app is runned as root, but changes about the owner are not expected when we use linux commands

Version of yq: 4.30.4
Operating system: linux
Installed via: binary release

**Command**
The command you ran:
```
./yq -i '.hosts[0] = "public.example.org"' /etc/ejabberd/ejabberd.yml
./yq -i 'with(.host_config."public.example.org"; . |= env(pgsql_connect) | ... style="")' /etc/ejabberd/ejabberd.yml
```

**Comment by mikefarah** [comment]
I can reproduce this - just not sure how to fix it - see https://github.com/mikefarah/yq/pull/1481

Any go experts?

**Comment by mikefarah** [comment]
Fixed in 4.30.7!
